src/common-result.ts

```diff
@@ -53,26 +56,23 @@ const CommonResultPrototype: CommonResult<unknown> = {
         return wrapper;
       }
 
-      return Promise.resolve()
-        .then(() => mapper(wrapper.value as any))
-        .then((newValue) => {
-          if (isResult(newValue)) {
-            return newValue.__value;
-          } else {
-            return resultWrapperFactory(newValue, false);
-          }
-        })
-        .catch((err) => resultWrapperFactory(err, true));
+      return wrapResultFactory(() => mapper(wrapper.value as any));
     });
     return commonResultFactory(newValWrapperPromise) as any;
   },
 
   unsafePromise() {
-    return this.__value.then((wrapper) =>
-      wrapper.isError
-        ? Promise.reject(wrapper.value)
-        : Promise.resolve(wrapper.value)
-    ) as any;
+    return this.__value.then((wrapper) => {
```

Comment:
I find this more readable, but move forward as you please

```
const wrapper = await this.__value;

if (!wrapper.isError) {
  return Promise.resolve(wrapper.value);
}

if (wrapper.value instanceof UnknownError) {
  return Promise.reject(wrapper.value.cause);
}

return Promise.reject(wrapper.value);
```

Reply:
I never use `await` in the library runnable code to have better control on final, build code shape (tests and examples are different story).
e.g. currently TSC build `await` to 
```ts
__awaiter(this, void 0, void 0, function* () {
            const test = yield this.__value;;
```
which is less obvious than `this.__value.then...`